python/dask_cudf/dask_cudf/backends.py

```diff
@@ -39,29 +39,34 @@ def concat_cudf(
     from dask.dataframe.utils import group_split_dispatch, hash_object_dispatch
     from cudf.core.column import column, CategoricalColumn, StringColumn
 
-    def _string_safe_hash(df):
+    def _handle_string(s):
+        if isinstance(s._column, StringColumn):
+            out_col = column.column_empty(len(s), dtype="int32", masked=False)
+            ptr = out_col.data.ptr
+            s._column.data_array_view.hash(devptr=ptr)
```

Comment:
doesn't `data_array_view` return a numba device array? What is this hash function?

Reply:
Hmm - It looks like we started using `._column.data.hash()` in #2994, and then @shwina changed it to `._column.data_array_view.hash()` in #3254.

I can follow up in a bit and make sure the hash is doing what we want/need.  If not, I'll figure out the *correct* solution.

Reply:
`data_array_view` seems to be returning an `nvstrings` object in this case.  So, this is indeed the same NVStrings-specific hashing function we were using before the refactor.